src/ws/cockpitauth.c

```diff
@@ -1241,13 +1241,38 @@ base64_decode_string (const char *enc)
   return dec;
 }
 
+static CockpitSession *
+session_for_cookie_value (CockpitAuth *self,
+                          const gchar *cookie_value)
+{
+  gchar *decoded = NULL;
+  const char *prefix = "v=2;k=";
+  CockpitSession *ret = NULL;
+
+  g_return_val_if_fail (self != NULL, FALSE);
+
+  if (cookie_value)
+    {
+      decoded = base64_decode_string (cookie_value);
```

Comment:
You can use `g_autofree` here, drop the declaration in line 1248 and the g_free below.

**update**: Oh, this is moved code. OK to keep if you prefer not to touch it.